src/main/java/bio/terra/flight/dataset/ingest/IngestSetupStep.java

```diff
@@ -56,6 +56,24 @@ public IngestSetupStep(DatasetService datasetService, BigQueryPdao bigQueryPdao)
 
     @Override
     public StepResult doStep(FlightContext context) {
+        IngestRequestModel ingestRequestModel = IngestUtils.getIngestRequestModel(context);
+        IngestUtils.GsUrlParts gsParts = IngestUtils.parseBlobUri(ingestRequestModel.getPath());
+
+        // FIXME: This check is preventing us from using wildcard patterns to ingest,
+        //  which is a giant PITA. How can we enable the pattern-matching without losing
+        //  the safety check?
+        //  https://cloud.google.com/bigquery/docs/loading-data-cloud-storage#load-wildcards
+        try {
+            Storage storage = StorageOptions.getDefaultInstance().getService();
+            BlobId blobId = BlobId.of(gsParts.getBucket(), gsParts.getPath());
+            Blob blob = storage.get(blobId);
+            if (!blob.exists()) { // NPE is here
```

Comment:
So since storage.get() can return a null, let's wrap it in an optional and check that as well here?

Reply:
Yup :+1: was trying to reproduce it in a test first, but it looks like it's not deterministic (???). If I can't figure it out by EOD I'll just go with the Optional wrapper & check